biz.aQute.launcher/src/aQute/launcher/Launcher.java

```diff
@@ -443,22 +444,30 @@ private int activate() throws Exception {
 				try {
 					Class<?> clazz = loader.loadClass((String) token);
 					BundleActivator activator = (BundleActivator) newInstance(clazz);
-					if (isImmediate(activator)) {
-						start(systemContext, result, activator);
+					if (isAfterBundleStart(activator)) {
+						result = start(systemContext, result, activator);
+					} else if (isBeforeBundleStart(activator)) {
+						startBeforeBundleStart.add(activator);
+					} else {
+						startAfterBundleStart.add(activator);
```

Comment:
So if it is not `isAfter` and not `isBefore` then you `startAfter`? That does not seem to read correctly.